tensorboard/webapp/runs/views/runs_table/runs_group_menu_button_component.ts

```diff
@@ -36,27 +36,37 @@ export class RunsGroupMenuButtonComponent {
   @Input() experimentIds!: string[];
   @Input() regexString!: string;
   @Input() selectedGroupBy!: GroupBy;
+  @Input() enableColorByExperiment!: boolean;
 
   @Output()
   onGroupByChange = new EventEmitter<GroupBy>();
 
   constructor(private readonly dialog: MatDialog) {}
 
   onRegexStringEdit() {
-    // data pass in the experiment id
-    const dialogRef = this.dialog.open(RegexEditDialogContainer, {
+    this.dialog.open(RegexEditDialogContainer, {
       maxHeight: '95vh',
       maxWidth: '80vw',
-      data: {experimentIds: this.experimentIds},
+      data: {
+        experimentIds: this.experimentIds,
+      },
     });
   }
 
   onGroupByRegexClick() {
     if (!this.regexString) {
       this.onRegexStringEdit();
     } else {
+      // If feature is enabled user can toggle between run and experiment name regex matching.
```

Comment:
IMO making this a toggle drifts too far from the original UX of this feature and from standard UX practices.

Clicking this item should just allow activating color by regex when another color method is selected (E.g. "Run" or "Experiment"):
![image](https://github.com/tensorflow/tensorboard/assets/736199/0935f963-1631-44eb-a38a-21de4587fc88)

Currently if "Regex" is already selected and you click "Regex", nothing happens - we should probably preserve this behavior.

Reply:
But aren't "Run" and "Experiment" GroupBy-s are totally different? I think changing the behavior of the regex depending on the previous groupby or other groupby will be unintuitive, if I understand you correctly.

Reply:
Discussed offline. Decided that when user clicks on regex the last choosen option will appear. If user wants to change the mode or the string, they have to press on pencil button.